Automattic/wp-calypso
# Stats: Fix tooltip flash on new sites

Related to:

https://github.com/Automattic/red-team/issues/192

## Proposed Changes

Guard against the `isNewSite` value being `null`. When that happens and the site is less than one week old, the value goes from `null` (falesy) to `true` causing the tooltip to show for a few seconds and then disappear. It's a bit jarring.

This fix guards against the unexpected `null` in the component logic. Ideally, we'd fix the logic in the `connect()` function to only pass boolean values. I'll look into that next.

## Why are these changes being made?


To prevent the Settings cog's tooltip flashing onscreen then hiding on new sites.

## Testing Instructions



Somewhat difficult to test as we don't have an easy way to reset the inputs.

1. Check out this branch at commit `b975196bdf68301f610d77ef84aa6b57ddca7994`. This commit contains code to force the issue.
2. Test with local Jetpack/Calypso environment to see the issue in action. Option 1 here: PCYsg-Pp7-p2
3. Move to commit `22220ea1ee44a76dd51674d6e9c28d4e3a7ebecf`. (debug conditions & fix are present)
4. Confirm the tooltip is no longer popping in and out.

## Pre-merge Checklist
